### ISSUE TITLE
single line return not emitting warning

### ISSUE THREAD

**Issue by binggh** [rust-lang/rust-clippy#8879]
### Summary

Expected to see a [`needless_return`](https://rust-lang.github.io/rust-clippy/master/index.html#needless_return) lint warning on a function that returned a single `format!()` line, yet clippy passed without failure despite using `cargo clippy --all-features --all-targets -- -D warnings`

Edit: on second thought it might not be a bug, but it feels like weird/unexpected behaviour seeing as there is also a lint for [`implicit_return`](clippy::implicit_return). How does clippy reconcile between these 2 lints?

### Reproducer

I tried this code:

```rust
fn needless_return() -> String {
  return format!("Hello {}", "world!");
}

fn main() {
  println!("{}", needless_return());
}
```

Expected a warning to remove the `return` and semicolon, yet no warnings were emitted from this short snippet.

### Version

```text
rustc 1.61.0 (fe5b13d68 2022-05-18)
binary: rustc
commit-hash: fe5b13d681f25ee6474be29d748c65adcd91f69e
commit-date: 2022-05-18
host: x86_64-apple-darwin
release: 1.61.0
LLVM version: 14.0.0
```

**Comment by Serial-ATA** [comment]
> How does clippy reconcile between these 2 lints?

`implicit_return` is a restriction lint that is allowed by default.

@rustbot label +I-false-negative +T-macros
